### PR TITLE
feat(core): make task runner selection configurable through env variable

### LINKED ISSUE #12138: Specify Runner Config with Environment Variable
- [x] I'd be willing to implement this feature ([contributing guide](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md))

## Description
I want to be able to specify a runner configuration (defined in `nx.json`) using an environment variable rather than the `--runner` command line option.

## Motivation
I am working on a project using nx, and the team wants to use a different runner configuration in CI than the default configuration our developers would use on their machines. This can be handled by passing the `--runner` option into the command line, but it makes for longer, less readable CI configuration. I would much rather set an environment variable and let nx pick that up.

## Suggested Implementation
Add logic so if the runner arg is not specified, and the environment variable `NX_RUNNER` is present, then set the runner option to the value of `NX_RUNNER`.

## Alternate Implementations
Any other idea I have come up with comes with grief for other users, and/or are unnecessarily complicated.

### PR DESCRIPTION
**Pull request by rv2673**
Have nx look at NX_RUNNER variable with lower priority for selecting task runner

## Current Behavior
Task runner can be selected by providing `--runner` argument with runner name, when running command. Otherwise `'default'` is used.

## Expected Behavior
Task runner can be selected by providing `--runner` argument with runner name, when running command. If not provided on command line looks if `NX_RUNNER` variable is present and uses its value if it is. Otherwise `'default'` is used.

## Related Issue(s)
Fixes #12138

### CODE REVIEW

**Review by AgentEnder — APPROVED** [review]

**Comment by rv2673** [comment]
@vsavkin @juristr can we get second review on this to get this merged?

**Comment by AgentEnder** [comment]
This one's just waiting for a docs review, from @juristr, @bcabanes, or @isaacplmann.

**Review by bcabanes — APPROVED** [review]

**Pull request merged**
